.github/workflows/run-tests.yaml

```
@@ -0,0 +1,25 @@
+name: Build and test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v4
```

The latest version of this action is `v4` (https://github.com/actions/checkout/releases).

I think we might as well go to the latest version of these third-party actions as we can, to extend as long as possible the time until they have to be upgraded again.

---

Updated to `actions/checkout@v4` and `actions/setup-python@v5`.